packages/runtime/container-runtime/src/summary/runningSummarizer.ts

```diff
@@ -91,21 +91,17 @@ export class RunningSummarizer extends TypedEventEmitter<ISummarizerEvents> impl
 			runtime,
 		);
 
-		// Before doing any heuristics or proceeding with its refreshing, if there is a summary ack received while
-		// this summarizer catches up, let's refresh state before proceeding with the summarization.
-		const lastAckRefSeq = await summarizer.handleSummaryAck();
-
-		await summarizer.waitStart();
-
 		// Handle summary acks asynchronously
 		// Note: no exceptions are thrown from processIncomingSummaryAcks handler as it handles all exceptions
-		summarizer.processIncomingSummaryAcks(lastAckRefSeq).catch((error) => {
+		summarizer.processIncomingSummaryAcks().catch((error) => {
 			createChildLogger({ logger }).sendErrorEvent(
 				{ eventName: "HandleSummaryAckFatalError" },
 				error,
 			);
 		});
 
+		await summarizer.waitStart();
```

Comment:
What do you think the impact of moving wait start to below processing incoming acks will do? Will it have a restart impact?

Reply:
Do you know what `waitStart` does? It doesn't immediately make sense to me when I read through the code

Reply:
Yeah, `waitStart` will wait for a summary ack if it sees a summary op that hasn't been acked. The idea being that if the previous summarizer closed before it saw an ack, this summarizer would do it.

It may need to happen before we summarize so that the summary doesn't fail. There are 3 cases where we see an op but not an ack:
- The previous summary succeeded but we loaded from the summary before that. In this case, if we don't process the ack, then the current summary will fail.
- The previous summary succeeded and we loaded from it. In this case, we don't need to process this ack and this change will ensure that.
- The previous summary failed. In this case, waiting for the ack just adds a delay but there is no way to know it.

Reply:
As for the impact, the kicking of summary heuristics and processing ack are both async and they race. Before this change, if there was a scenario like case # 1 above, chances are that we wouid start a summarization before processing the ack and the summary would fail. Now, we will process the ack first and that will fail which is cheaper and faster that doing the summary and failling.

Reply:
I actually reverted back the order. Also, added back synchronously processing the first ack but only if its newer that the one the client loaded from. Added a comment to explain why it is needed.

Reply:
Just took a look, looks good